src/main/java/org/owasp/esapi/Encoder.java

```diff
@@ -107,16 +109,42 @@
  * in other applications that could access that data in the database.
  * </li>
  * <li><b>Avoiding multiple <i>nested</i> contexts:</b>
- * A really tricky situation to get correct is hen there are multiple nested
+ * A really tricky situation to get correct is when there are multiple nested
  * encoding contexts. But far, the most common place this seems to come up is
- * untrusted URLs used in JavaScript. How should you handle that? Well, to be
- * honest, the best way is to rewrite your code to avoid it.  An example of
+ * untrusted URLs used in JavaScript. How should you handle that? Well,
+ * the best way is to rewrite your code to avoid it!  An example of
  * this that is well worth reading may be found at
  * <a href="https://lists.owasp.org/pipermail/esapi-dev/2012-March/002090"
  * target="_blank" rel="noopener noreferrer">ESAPI-DEV mailing list archives:
  * URL encoding within JavaScript</a>. Be sure to read the entire thread.
  * The question itself is too nuanced to be answered in Javadoc, but now,
- * hopefully you are at least aware of the potential pitfalls.
+ * hopefully you are at least aware of the potential pitfalls. There is little
+ * avaiable research or examples on how to do output encodeing when multiple
+ * mixed encodings are required, although one that you may find useful is
+ * <a href="https://arxiv.org/pdf/1804.01862.pdf" target="_blank"
+ * rel="noopener noreferrer">
+ * Automated Detecting and Repair of Cross-SiteScripting Vulnerabilities through Unit Testing</a>
+ * It at least discusses a few of the common errors involved in multiple mixed
+ * encoding contexts.
+ * </li><li><b>A word about unit testing:</b>
+ * Unit testing this is hard. You may be satisfied with stopped after you have
+ * tested against the ubiquitous XSS test case of
+ * <pre>
+ *      &lt;/script&gt;alert(1)&lt;/script&gt;
+ * </pre>
+ * or similiar simplistic XSS attack payloads and if that is properly encoded
+ * (or, you don't see an alert box popped in your browser), you consider it
+ * "problem fixed", consider the unit testing sufficient. Unfortunately, that
+ * minimalist testing may not always detect places were you used the wrong decoder. You need to do better.
+ * Fortunately, the aforementioned link,
+ * <a href="https://arxiv.org/pdf/1804.01862.pdf" target="_blank"
+ * rel="noopener noreferrer">
+ * Automated Detecting and Repair of Cross-SiteScripting Vulnerabilities through Unit Testing</a>
+ * provides some insight. You may also wish to look at the
+ * <a href="https://github.com/ESAPI/esapi-java-legacy/blob/develop/src/test/java/org/owasp/esapi/reference/EncoderTest.java"
+ * target="_blank" rel="noopener noreferrer">ESAPI Encoder JUnittest cases</a>.
+ * If you are really ambitious, an excellent resource for XSS attack patters is
```

Comment:
Should we also discuss SecLists and FuzzDB?

Reply:
Most of this Encoder Javadoc originated on my running across https://arxiv.org/pdf/1810.01017.pdf, which was a usability study on the ESAPI Encoder. I have been interacting with the 2 authors on Google Drive commenting on their paper and trying to gain insight as to how to improve ESAPI Encoder to make it more noob friendly. They had remarked that students had no idea how to test to see if then had secured their code correctly, thus these specific additions. So that is the target audience with these javadoc comments on testing.